common.go

```diff
@@ -453,9 +453,11 @@ func GetLanguageByAlias(alias string) (lang string, ok bool) {
 // GetLanguagesByGitAttributes returns either a string slice with the lenguage if the filename match with a regExp in .gitattributes
 //or return nil in case of none regexp matchs the filename . It complies with the signature to be a Strategy type.
 func GetLanguagesByGitAttributes(filename string, content []byte, candidates []string) []string {
-	for regExp, language := range languageGitAttributes {
-		if regExp.MatchString(filename) {
-			return []string{language}
+	if loadedGitAttributes != nil {
```

Comment:
I don't think having a global variable floating around different files is a good idea.